salt/report_upload/srv/salt/_runners/foreman_report_upload.py

```diff
@@ -100,7 +125,10 @@ def now(highstate):
 
     try:
         report = create_report(base64.b64decode(highstate))
+        get_lock()
         upload(report)
+        release_lock()
     except Exception as exc:
+        release_lock()
```

Comment:
I would put the release_lock in a finally block.

Reply:
Good hint, done.